qiskit/providers/backend.py

```diff
@@ -229,3 +240,435 @@ class can handle either situation.
+            if prop.t1 is None:
+                raise ValueError(f"T1 is not defined for qubit {qubits[index]}")
+            out_list.append(prop.t1)
+        return np.array(out_list)
+
+    def t2(self, qubits: Union[int, List[int]]) -> Union[float, np.array]:
+        """Return the T2 time of a given qubit
+
+        Args:
+            qubits: The qubit index or indices to get the T2 time for. If
+                a list is specified the output will be a list with the
+                T2 time for the specified qubits in the same order.
+
+        Returns:
+            t2: the T2 time for the specified qubit(s) in seconds
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                qubit properties for a qubit
+            ValueError: if T2 is not defined for the specified qubit
+        """
+        props = self.qubits(qubits)
+        if isinstance(props, QubitProperties):
+            if props.t2 is None:
+                raise ValueError(f"T2 is not defined for qubit {qubits}")
+            return props.t1
+        out_list = []
+        for index, prop in enumerate(props):
+            if prop.t2 is None:
+                raise ValueError(f"T2 is not defined for qubit {qubits[index]}")
+            out_list.append(prop.t2)
+        return np.array(out_list)
+
+    @property
+    def dt(self) -> Union[float, None]:
+        """Return the system time resolution of input signals
+
+        This is required to be implemented if the backend supports Pulse
+        scheduling.
+
+        Returns:
+            dt: The input signal timestep in seconds. If the backend doesn't
+            define ``dt`` ``None`` will be returned
+        """
+        return self.target.dt
+
+    @property
+    def dtm(self) -> float:
+        """Return the system time resolution of output signals
+
+        Returns:
+            dtm: The output signal timestep in seconds.
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                output signal timestep
+        """
+        raise NotImplementedError
+
+    @property
+    def meas_map(self) -> List[List[int]]:
+        """Return the grouping of measurements which are multiplexed
+
+        This is required to be implemented if the backend supports Pulse
+        scheduling.
+
+        Returns:
+            meas_map: The grouping of measurements which are multiplexed
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                measurement mapping
+        """
+        raise NotImplementedError
+
+    @property
+    def instruction_schedule_map(self):
+        """Return the :class:`~qiskit.pulse.InstructionScheduleMap` for the
+        instructions defined in this backend's target."""
+        return self.target.instruction_schedule_map()
+
+    def qubits(self, qubit: Union[int, List[int]]) -> Union[QubitProperties, List[QubitProperties]]:
```

Comment:
@ajavadia I'm thinking of renaming this `qubit_properties` because we're not returning a `Qubit` but the `QubitProperties` for a given qubit. Does that work for you?

Reply:
I renamed this in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/0d1b2dc84e33e7cf1d6124d3ac14414d1d8cd62c we can revert this if there is an objection though.